src/app/search/search.component.ts

```diff
@@ -516,6 +526,16 @@ export class SearchComponent implements OnInit, OnDestroy {
     this.sortModeMap.get(category).SortBy = sortMode;
   }
 
+  // Get autocomplete terms
+  onFacetSearchKey(key) {
+    const values = this.facetSearchText.toLowerCase();
+    const unfilteredItems = Array.from(this.orderedBuckets.get(key).Items.entries());
+    const filteredItems = unfilteredItems.filter((item) => item[0].toLowerCase().includes(values)).map((item) => item[0]);
+    if (filteredItems) {
```

Comment:
This will always be true. If you want the logic this seems to imply, you probably need to test that it's empty.

That said, what happens if it is empty? Do you not want to clear the existing terms (I don't know)?

Reply:
Oops, my logic went awry here. If its empty, the function setFacetAutocompleteTerms can still be called but will leave autocomplete array empty